bin/azure-deploy

```diff
@@ -2,16 +2,29 @@
 set -e
 
 ENVIRONMENT_NAME=$1
-DEPLOYMENT_NAME="core"
+SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
+DEPLOYMENT_NAME="app"
 RESOURCE_LOCATION="West Europe"
 TEMPLATE_FILE_PATH="azure/template.json"
 PARAMETERS_FILE_PATH="azure/parameters/$ENVIRONMENT_NAME.json"
 
+if [[ -z "$SUBSCRIPTION_ID" ]]; then
+	echo "Your subscription ID can be looked up with the CLI using: az account show --out json "
+	echo "Enter your subscription ID:"
+	read SUBSCRIPTION_ID
+	[[ "${SUBSCRIPTION_ID:?}" ]]
+fi
+
+if [[ -z "$RESOURCE_GROUP_NAME" ]]; then
+	echo "Enter your resource group name:"
+	read RESOURCE_GROUP_NAME
+	[[ "${RESOURCE_GROUP_NAME:?}" ]]
+fi
```

Comment:
What motivated this change? I imagined we'd hardcode the values for the different environments into the `case` statement below, rather than requiring extra setup from users.

Reply:
Ok, I now see what motivated it. I don't really see how this improves security, as if you have access to the subscription or the resource group, you can easily query them with the CLI. We're not gaining anything useful, and we're adding (unnecessary IMO) complication to new developer setup.

Reply:
We're also hardcoding the KeyVault resource group.

Reply:
I think we need a better solution that environment variables, as these will be different for each environment, and requiring the user to manually update them when switching environments is error prone. What are we actually gaining from taking them out of the script?

Reply:
For context: https://stackoverflow.com/a/45663968

Reply:
Yep, I think we can safely leave these in. It will make deployment much easier and I don't think there's any real risk in having these values open.